board/opentrons/ot2/kernelRamfs/rootfs-overlay/init

```diff
@@ -0,0 +1,78 @@
+#!/bin/sh
+
+###################
+# Opentrons KernelRamFS     
+###################
+#
+# The purpose of this init script is to allow
+# one stage updates when updating from buildroot
+# with an old kernel to buildroot with new kernel.
+# 
+# The reason why we need to do that is because the device tree (dtb)
+# and device tree overlays (dtbo) files have changed and so a clean
+# update causes the device to get bricked needing a reflash.
+#
+# The way we accomplish this is by shipping the rootfs with this kernelRamfs
+# as the zImage in the /boot dir, as well as the actual kernel image.
+# When the rootfs is written in the inactive partition and the system
+# restarts we boot into this kernelRamfs. From here we mount the active
+# rootfs partition, copy the dtb(s) + overlays + config.txt files from
+# /boot to the boot partition and rename the kernel. We then set the 
+# uboot env variable 'ot2part' to the current part and reboot. Once 
+# the system reboots it will boot into the new kernel.
+###################
+
+echo "Welcome to Opentrons kernelRamfs"
+
+function recovery {
+   # drop into shell if something goes wrong
+   echo "Something went wrong, entering recovery..."
+   setsid sh -c 'exec sh </dev/tty1 >/dev/tty1 2>&1'
+}
+
+echo "Setting up ramdisk"
+# Mount the /proc, /sys and /dev filesystems.
+/bin/mount -t proc none /proc || recovery
```

Comment:
might be better to `set -eo pipefail` and `trap recovery EXIT`, saves you from having to do `|| recovery` all over the place (and saves you from forgetting to do that on some random line)